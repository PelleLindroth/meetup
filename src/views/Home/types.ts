import { Meetup } from '../../db/models/Meetup'

export type HomeProps = {
  meetups: Meetup[]
  setMeetups: React.Dispatch<React.SetStateAction<Meetup[]>>
}
