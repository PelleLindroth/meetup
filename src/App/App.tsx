import { useEffect, useState } from 'react'
import { Routes, Route } from 'react-router-dom'
import { getAllMeetups, getStoredUser } from '../db'
import { Meetup } from '../db/models/Meetup'
import { User } from '../db/models/User'
import DateContextProvider from '../contexts/DateContext'
import Header from '../Header'
import { Home, Login, SingleMeetup, CreateMeetup, Profile } from '../views'
import styles from './App.module.scss'

function App() {
  const [meetups, setMeetups] = useState<Meetup[]>([])
  const [user, setUser] = useState<User | null>(null)

  useEffect(() => {
    setMeetups(getAllMeetups())
    setUser(getStoredUser())
  }, [])

  return (
    <div className={styles.wrapper}>
      <DateContextProvider>
        <Header user={user} setUser={setUser} />
        <Routes>
          <Route
            path="/"
            element={<Home meetups={meetups} setMeetups={setMeetups} />}
          />
          <Route
            path="/meetup/:id"
            element={<SingleMeetup meetups={meetups} user={user} />}
          />
          {!user && (
            <Route path="/login" element={<Login setUser={setUser} />} />
          )}
          {user && (
            <Route
              path="/profile/:id"
              element={<Profile meetups={meetups} />}
            />
          )}
          {user && (
            <Route
              path="/create"
              element={
                <CreateMeetup
                  user={user}
                  meetups={meetups}
                  setMeetups={setMeetups}
                />
              }
            />
          )}
        </Routes>
      </DateContextProvider>
    </div>
  )
}

export default App
