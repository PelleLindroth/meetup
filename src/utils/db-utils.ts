import { Meetup } from '../db/models/Meetup'

export const parseDates = (meetups: Meetup[]) => {
  meetups.forEach((meetup) => {
    meetup.date = new Date(meetup.date)
    meetup.comments.forEach((comment) => {
      comment.submittedAt = new Date(comment.submittedAt)
    })
  })
}

export const sortMeetupsChronologically = (meetups: Meetup[]) => {
  meetups.sort((a, b) => a.date.getTime() - b.date.getTime())
}
