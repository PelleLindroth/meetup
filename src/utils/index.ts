import { User } from '../db/users'
import { Meetup } from '../db/meetups'

export const formatDate = (date: Date) => {
  return `${new Intl.DateTimeFormat('en-GB', {
    year: 'numeric',
    month: 'long',
    day: '2-digit',
    hour: 'numeric',
    minute: 'numeric',
  }).format(date)}`
}

const sortChronologically = (array: Meetup[]) => {
  return array.sort((a, b) => a.date.getTime() - b.date.getTime())
}

export const createMeetupList = (
  meetups: Meetup[],
  listType: string,
  searchPhrase: string,
  searchFilter: string
) => {
  const meetupList = meetups.filter((meetup) => {
    if (listType === 'past') {
      return (
        meetup.date.getTime() < Date.now() &&
        meetup.title.toLowerCase().includes(searchPhrase.toLowerCase()) &&
        (searchFilter === listType || searchFilter === 'all')
      )
    } else {
      return (
        meetup.date.getTime() > Date.now() &&
        meetup.title.toLowerCase().includes(searchPhrase.toLowerCase()) &&
        (searchFilter === listType || searchFilter === 'all')
      )
    }
  })

  return sortChronologically(meetupList)
}

export const createUserLists = (user: User, meetups: Meetup[]) => {
  return {
    ownUpcoming: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          meetup.arranger.id === user!.id &&
          meetup.date.getTime() > new Date().getTime()
      )
    ),
    ownPast: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          meetup.arranger.id === user!.id &&
          meetup.date.getTime() < new Date().getTime()
      )
    ),
    upcomingAttending: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          user!.attending.includes(meetup.id) &&
          meetup.date.getTime() > new Date().getTime()
      )
    ),
    pastAttended: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          user!.attending.includes(meetup.id) &&
          meetup.date.getTime() < new Date().getTime()
      )
    ),
  }
}
