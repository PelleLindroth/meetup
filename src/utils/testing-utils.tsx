import { render } from '@testing-library/react'
import { ReactElement } from 'react'
import { MemoryRouter } from 'react-router-dom'
import { Routes, Route } from 'react-router'

export const renderWithRouter = (ui: ReactElement, options?: any) =>
  render(ui, { wrapper: MemoryRouter, ...options })

export const renderWithPath = (ui: ReactElement, entry: string, path: string) =>
  render(
    <MemoryRouter initialEntries={[entry]}>
      <Routes>
        <Route path={path} element={ui} />
      </Routes>
    </MemoryRouter>
  )
