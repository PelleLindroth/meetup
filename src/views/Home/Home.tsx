import { useContext, useState } from 'react'
import { createMeetupList, list } from '../../utils'
import { DateContext } from '../../contexts/DateContext'
import SearchArea from './components/SearchArea/SearchArea'
import MeetupList from './components/MeetupList'
import { HomeProps } from './types'

const Home = (props: HomeProps) => {
  const { meetups } = props
  const { customDate } = useContext(DateContext)!
  const [searchPhrase, setSearchPhrase] = useState<string>('')
  const [searchFilter, setSearchFilter] = useState<list>('all')

  return (
    <>
      <SearchArea
        searchPhrase={searchPhrase}
        setSearchPhrase={setSearchPhrase}
        searchFilter={searchFilter}
        setSearchFilter={setSearchFilter}
      />
      <section>
        {(searchFilter === 'upcoming' || searchFilter === 'all') && (
          <h2>UPCOMING EVENTS</h2>
        )}
        <MeetupList
          meetups={createMeetupList(
            meetups,
            'upcoming',
            searchPhrase,
            searchFilter,
            customDate
          )}
          upcoming
        />
        {(searchFilter === 'past' || searchFilter === 'all') && (
          <h2>PAST EVENTS</h2>
        )}

        <MeetupList
          meetups={createMeetupList(
            meetups,
            'past',
            searchPhrase,
            searchFilter,
            customDate
          )}
        />
      </section>
    </>
  )
}

export default Home
