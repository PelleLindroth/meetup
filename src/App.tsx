import { useEffect, useState } from 'react'
import { Routes, Route } from 'react-router-dom'
import { Meetup } from './db/models/Meetup'
import { getAllMeetups, getStoredUser } from './db'
import DateContextProvider from './contexts/DateContext'
import Header from './Header'
import Home from './views/Home'
import Login from './views/Login'
import SingleMeetup from './views/SingleMeetup'
import CreateMeetup from './views/CreateMeetup'
import Profile from './views/Profile'
import { UserImpl } from './db/models/User'
import styles from './App/App.module.scss'

function App() {
  const [meetups, setMeetups] = useState<Meetup[]>([])
  const [user, setUser] = useState<UserImpl | null>(null)

  useEffect(() => {
    const dbMeetups = getAllMeetups()

    setMeetups(dbMeetups)

    const user = getStoredUser()

    setUser(user)
  }, [])

  return (
    <div className={styles.wrapper}>
      <DateContextProvider>
        <Header user={user} setUser={setUser} />
        <Routes>
          <Route
            path="/"
            element={<Home meetups={meetups} setMeetups={setMeetups} />}
          />
          <Route
            path="/meetup/:id"
            element={<SingleMeetup meetups={meetups} user={user} />}
          />
          {!user && (
            <Route path="/login" element={<Login setUser={setUser} />} />
          )}
          {user && (
            <Route
              path="/profile/:id"
              element={<Profile meetups={meetups} />}
            />
          )}
          {user && (
            <Route
              path="/create"
              element={
                <CreateMeetup
                  user={user}
                  meetups={meetups}
                  setMeetups={setMeetups}
                />
              }
            />
          )}
        </Routes>
      </DateContextProvider>
    </div>
  )
}

export default App
