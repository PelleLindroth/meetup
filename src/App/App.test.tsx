import { mount } from 'enzyme'
import { screen } from '@testing-library/react'
import { renderWithRouter } from '../utils/testing-utils'
import { MemoryRouter } from 'react-router'
import App from '../App'
import Home from '../views/Home'
import userEvent from '@testing-library/user-event'

describe('App unit tests', () => {
  it('renders App correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )

    expect(wrapper.find(App)).toMatchSnapshot()
  })
  it('renders a Header with logo and login button', () => {
    renderWithRouter(<App />)

    const header = screen.getByRole('banner')
    const logo = screen.getByAltText('logo')
    const loginButton = screen.getByRole('button', { name: /log in/i })

    expect(header).toBeInTheDocument()
    expect(logo).toBeInTheDocument()
    expect(loginButton).toBeInTheDocument()
  })
  it('initially renders Home view with 2 lists with a total of 4 meetups', () => {
    const wrapper = mount(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )

    expect(wrapper.find(Home).find('ul')).toHaveLength(2)
    expect(wrapper.find(Home).find('li')).toHaveLength(4)
  })
})

describe('App integration tests', () => {
  it('renders login view when log in button is clicked', async () => {
    renderWithRouter(<App />)

    const loginButton = screen.getByRole('button', { name: /log in/i })

    userEvent.click(loginButton)

    const loginForm = await screen.findByRole('form')

    expect(loginForm).toBeInTheDocument()
  })
})

// App

// Home view
// shows meetup details on separate page when clicking arrow link
// renders existing comments and reviews on past events on detail page
