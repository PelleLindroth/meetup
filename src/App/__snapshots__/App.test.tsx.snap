// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit tests renders App correctly 1`] = `
<App>
  <div
    className="wrapper"
  >
    <Header
      setUser={[Function]}
      user={null}
    >
      <header
        className="header"
      >
        <div
          className="innerContainer"
        >
          <Link
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                src="logo.png"
              />
            </a>
          </Link>
          <Link
            to="/login"
          >
            <a
              href="/login"
              onClick={[Function]}
            >
              <button
                className="loginButton"
              >
                Log in
              </button>
            </a>
          </Link>
        </div>
      </header>
    </Header>
    <Routes>
      <Home
        meetups={
          Array [
            Object {
              "arranger": Object {
                "attending": Array [
                  "1",
                  "2",
                ],
                "email": "tjalle@yahoo.com",
                "firstName": "Tjalle",
                "id": "1",
                "lastName": "Bergkvist",
                "password": "grillkorv",
                "reviewed": Array [
                  "1",
                ],
              },
              "attending": 54,
              "capacity": null,
              "comments": Array [
                Object {
                  "id": "1",
                  "submittedAt": 2021-12-16T14:58:33.837Z,
                  "text": "Great day!",
                  "userId": "2",
                },
                Object {
                  "id": "2",
                  "submittedAt": 2021-12-16T14:59:22.526Z,
                  "text": "Yes, it was, thanks!",
                  "userId": "1",
                },
                Object {
                  "id": "3",
                  "submittedAt": 2021-12-17T12:34:42.949Z,
                  "text": "Damn, I missed this",
                  "userId": null,
                },
              ],
              "date": 2021-08-28T17:00:00.000Z,
              "description": "We meet in the park with pizzas and hang out all night",
              "id": "1",
              "location": Object {
                "city": "Stockholm",
                "street": "Vitabergsparken",
              },
              "online": false,
              "reviews": Array [
                Object {
                  "id": "1",
                  "meetupId": "1",
                  "rating": 4,
                },
                Object {
                  "id": "2",
                  "meetupId": "1",
                  "rating": 5,
                },
              ],
              "title": "Pizza Picnic",
            },
            Object {
              "arranger": Object {
                "attending": Array [],
                "email": "babsan@yahoo.com",
                "firstName": "Barbro",
                "id": "3",
                "lastName": "Lundborg",
                "password": "password123",
                "reviewed": Array [],
              },
              "attending": 356,
              "capacity": 500,
              "comments": Array [],
              "date": 2021-10-31T18:00:00.000Z,
              "description": "Halloween party in the street, all night long",
              "id": "5",
              "location": Object {
                "city": "Stockholm",
                "street": "Strandvägen",
              },
              "online": false,
              "reviews": Array [],
              "title": "Halloween Street Party",
            },
            Object {
              "arranger": Object {
                "attending": Array [],
                "email": "babsan@yahoo.com",
                "firstName": "Barbro",
                "id": "3",
                "lastName": "Lundborg",
                "password": "password123",
                "reviewed": Array [],
              },
              "attending": 234,
              "capacity": null,
              "comments": Array [],
              "date": 2022-01-10T10:00:00.000Z,
              "description": "Learn a new programming language from the ground up",
              "id": "2",
              "online": true,
              "reviews": Array [],
              "title": "Rust course",
              "url": "https://www.udemy.com/course/rust-lang/",
            },
            Object {
              "arranger": Object {
                "attending": Array [
                  "1",
                ],
                "email": "kenta@yahoo.com",
                "firstName": "Kenta",
                "id": "2",
                "lastName": "Andersson",
                "password": "bananpaj",
                "reviewed": Array [
                  "1",
                ],
              },
              "attending": 23478,
              "capacity": 43883,
              "comments": Array [],
              "date": 2022-05-13T16:00:00.000Z,
              "description": "A classic football match between the south and the north",
              "id": "3",
              "location": Object {
                "city": "41005 Sevilla",
                "street": "C. Sevilla Fútbol Club",
              },
              "online": false,
              "reviews": Array [],
              "title": "Sevilla - Real Sociedad",
            },
            Object {
              "arranger": Object {
                "attending": Array [
                  "1",
                  "2",
                ],
                "email": "tjalle@yahoo.com",
                "firstName": "Tjalle",
                "id": "1",
                "lastName": "Bergkvist",
                "password": "grillkorv",
                "reviewed": Array [
                  "1",
                ],
              },
              "attending": 126,
              "capacity": null,
              "comments": Array [],
              "date": 2022-07-02T15:00:00.000Z,
              "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
              "id": "4",
              "location": Object {
                "city": "Stockholm",
                "street": "Gärdet",
              },
              "online": false,
              "reviews": Array [],
              "title": "Kite festival",
            },
          ]
        }
        setMeetups={[Function]}
      >
        <SearchArea
          searchFilter="all"
          searchPhrase=""
          setSearchFilter={[Function]}
          setSearchPhrase={[Function]}
        >
          <section
            className="searchArea"
          >
            <div
              className="searchBox"
            >
              <div
                className="imageWrapper"
              >
                <img
                  alt="Search icon"
                  src="search.png"
                />
              </div>
              <input
                onChange={[Function]}
                placeholder="What do you want to do?"
                type="search"
                value=""
              />
            </div>
            <select
              id="search-filter"
              name="search-filter"
              onChange={[Function]}
              value="all"
            >
              <option
                value="all"
              >
                Search all events
              </option>
              <option
                value="upcoming"
              >
                Search upcoming events
              </option>
              <option
                value="past"
              >
                Search past events
              </option>
            </select>
          </section>
        </SearchArea>
        <section>
          <h2>
            UPCOMING EVENTS
          </h2>
          <MeetupList
            meetups={
              Array [
                Object {
                  "arranger": Object {
                    "attending": Array [],
                    "email": "babsan@yahoo.com",
                    "firstName": "Barbro",
                    "id": "3",
                    "lastName": "Lundborg",
                    "password": "password123",
                    "reviewed": Array [],
                  },
                  "attending": 234,
                  "capacity": null,
                  "comments": Array [],
                  "date": 2022-01-10T10:00:00.000Z,
                  "description": "Learn a new programming language from the ground up",
                  "id": "2",
                  "online": true,
                  "reviews": Array [],
                  "title": "Rust course",
                  "url": "https://www.udemy.com/course/rust-lang/",
                },
                Object {
                  "arranger": Object {
                    "attending": Array [
                      "1",
                    ],
                    "email": "kenta@yahoo.com",
                    "firstName": "Kenta",
                    "id": "2",
                    "lastName": "Andersson",
                    "password": "bananpaj",
                    "reviewed": Array [
                      "1",
                    ],
                  },
                  "attending": 23478,
                  "capacity": 43883,
                  "comments": Array [],
                  "date": 2022-05-13T16:00:00.000Z,
                  "description": "A classic football match between the south and the north",
                  "id": "3",
                  "location": Object {
                    "city": "41005 Sevilla",
                    "street": "C. Sevilla Fútbol Club",
                  },
                  "online": false,
                  "reviews": Array [],
                  "title": "Sevilla - Real Sociedad",
                },
                Object {
                  "arranger": Object {
                    "attending": Array [
                      "1",
                      "2",
                    ],
                    "email": "tjalle@yahoo.com",
                    "firstName": "Tjalle",
                    "id": "1",
                    "lastName": "Bergkvist",
                    "password": "grillkorv",
                    "reviewed": Array [
                      "1",
                    ],
                  },
                  "attending": 126,
                  "capacity": null,
                  "comments": Array [],
                  "date": 2022-07-02T15:00:00.000Z,
                  "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
                  "id": "4",
                  "location": Object {
                    "city": "Stockholm",
                    "street": "Gärdet",
                  },
                  "online": false,
                  "reviews": Array [],
                  "title": "Kite festival",
                },
              ]
            }
            upcoming={true}
          >
            <ul
              className="meetupList"
              data-testid="upcoming-events"
            >
              <MeetupCard
                key="2"
                meetup={
                  Object {
                    "arranger": Object {
                      "attending": Array [],
                      "email": "babsan@yahoo.com",
                      "firstName": "Barbro",
                      "id": "3",
                      "lastName": "Lundborg",
                      "password": "password123",
                      "reviewed": Array [],
                    },
                    "attending": 234,
                    "capacity": null,
                    "comments": Array [],
                    "date": 2022-01-10T10:00:00.000Z,
                    "description": "Learn a new programming language from the ground up",
                    "id": "2",
                    "online": true,
                    "reviews": Array [],
                    "title": "Rust course",
                    "url": "https://www.udemy.com/course/rust-lang/",
                  }
                }
                upcoming={true}
              >
                <li
                  aria-label="meetup-card"
                  className="cardWrapper"
                >
                  <div
                    className="cardHeader"
                    style={
                      Object {
                        "backgroundColor": "#0E81C9",
                      }
                    }
                    title="upcoming-event-header"
                  >
                    <h2>
                      Rust course
                    </h2>
                  </div>
                  <div
                    className="cardBody"
                  >
                    <div
                      className="leftColumn"
                    >
                      <div
                        className="dateRow"
                      >
                        <img
                          alt="Time icon"
                          className="icon"
                          src="time.png"
                        />
                        <h3
                          title="date"
                        >
                          10 January 2022, 11:00
                           
                          CET
                        </h3>
                      </div>
                      <div
                        className="locationRow"
                      >
                        <img
                          alt="Cloud icon"
                          className="icon"
                          src="cloud.png"
                        />
                        <h3
                          title="location"
                        >
                          Online event
                        </h3>
                      </div>
                      <p
                        className="capacityRow"
                        title="capacity"
                      >
                        <strong>
                          Capacity: 
                        </strong>
                        <span
                          className="available"
                        >
                          Unlimited
                        </span>
                      </p>
                    </div>
                    <div
                      className="rightColumn"
                    >
                      <Link
                        to="/meetup/2"
                      >
                        <a
                          href="/meetup/2"
                          onClick={[Function]}
                        >
                          <img
                            alt="Arrow icon"
                            className="arrowIcon"
                            src="arrow.png"
                            title="See meetup details"
                          />
                        </a>
                      </Link>
                    </div>
                  </div>
                </li>
              </MeetupCard>
              <MeetupCard
                key="3"
                meetup={
                  Object {
                    "arranger": Object {
                      "attending": Array [
                        "1",
                      ],
                      "email": "kenta@yahoo.com",
                      "firstName": "Kenta",
                      "id": "2",
                      "lastName": "Andersson",
                      "password": "bananpaj",
                      "reviewed": Array [
                        "1",
                      ],
                    },
                    "attending": 23478,
                    "capacity": 43883,
                    "comments": Array [],
                    "date": 2022-05-13T16:00:00.000Z,
                    "description": "A classic football match between the south and the north",
                    "id": "3",
                    "location": Object {
                      "city": "41005 Sevilla",
                      "street": "C. Sevilla Fútbol Club",
                    },
                    "online": false,
                    "reviews": Array [],
                    "title": "Sevilla - Real Sociedad",
                  }
                }
                upcoming={true}
              >
                <li
                  aria-label="meetup-card"
                  className="cardWrapper"
                >
                  <div
                    className="cardHeader"
                    style={
                      Object {
                        "backgroundColor": "#0E81C9",
                      }
                    }
                    title="upcoming-event-header"
                  >
                    <h2>
                      Sevilla - Real Sociedad
                    </h2>
                  </div>
                  <div
                    className="cardBody"
                  >
                    <div
                      className="leftColumn"
                    >
                      <div
                        className="dateRow"
                      >
                        <img
                          alt="Time icon"
                          className="icon"
                          src="time.png"
                        />
                        <h3
                          title="date"
                        >
                          13 May 2022, 18:00
                           
                          CET
                        </h3>
                      </div>
                      <div
                        className="locationRow"
                      >
                        <img
                          alt="Map pin icon"
                          className="icon"
                          src="pin.png"
                        />
                        <h3
                          title="location"
                        >
                          C. Sevilla Fútbol Club, 41005 Sevilla
                        </h3>
                      </div>
                      <p
                        className="capacityRow"
                        title="capacity"
                      >
                        <strong>
                          Capacity: 
                        </strong>
                        <span>
                          43883, 
                        </span>
                        <span
                          className="available"
                        >
                          20405 available
                        </span>
                      </p>
                    </div>
                    <div
                      className="rightColumn"
                    >
                      <Link
                        to="/meetup/3"
                      >
                        <a
                          href="/meetup/3"
                          onClick={[Function]}
                        >
                          <img
                            alt="Arrow icon"
                            className="arrowIcon"
                            src="arrow.png"
                            title="See meetup details"
                          />
                        </a>
                      </Link>
                    </div>
                  </div>
                </li>
              </MeetupCard>
              <MeetupCard
                key="4"
                meetup={
                  Object {
                    "arranger": Object {
                      "attending": Array [
                        "1",
                        "2",
                      ],
                      "email": "tjalle@yahoo.com",
                      "firstName": "Tjalle",
                      "id": "1",
                      "lastName": "Bergkvist",
                      "password": "grillkorv",
                      "reviewed": Array [
                        "1",
                      ],
                    },
                    "attending": 126,
                    "capacity": null,
                    "comments": Array [],
                    "date": 2022-07-02T15:00:00.000Z,
                    "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
                    "id": "4",
                    "location": Object {
                      "city": "Stockholm",
                      "street": "Gärdet",
                    },
                    "online": false,
                    "reviews": Array [],
                    "title": "Kite festival",
                  }
                }
                upcoming={true}
              >
                <li
                  aria-label="meetup-card"
                  className="cardWrapper"
                >
                  <div
                    className="cardHeader"
                    style={
                      Object {
                        "backgroundColor": "#0E81C9",
                      }
                    }
                    title="upcoming-event-header"
                  >
                    <h2>
                      Kite festival
                    </h2>
                  </div>
                  <div
                    className="cardBody"
                  >
                    <div
                      className="leftColumn"
                    >
                      <div
                        className="dateRow"
                      >
                        <img
                          alt="Time icon"
                          className="icon"
                          src="time.png"
                        />
                        <h3
                          title="date"
                        >
                          02 July 2022, 17:00
                           
                          CET
                        </h3>
                      </div>
                      <div
                        className="locationRow"
                      >
                        <img
                          alt="Map pin icon"
                          className="icon"
                          src="pin.png"
                        />
                        <h3
                          title="location"
                        >
                          Gärdet, Stockholm
                        </h3>
                      </div>
                      <p
                        className="capacityRow"
                        title="capacity"
                      >
                        <strong>
                          Capacity: 
                        </strong>
                        <span
                          className="available"
                        >
                          Unlimited
                        </span>
                      </p>
                    </div>
                    <div
                      className="rightColumn"
                    >
                      <Link
                        to="/meetup/4"
                      >
                        <a
                          href="/meetup/4"
                          onClick={[Function]}
                        >
                          <img
                            alt="Arrow icon"
                            className="arrowIcon"
                            src="arrow.png"
                            title="See meetup details"
                          />
                        </a>
                      </Link>
                    </div>
                  </div>
                </li>
              </MeetupCard>
            </ul>
          </MeetupList>
          <h2>
            PAST EVENTS
          </h2>
          <MeetupList
            meetups={
              Array [
                Object {
                  "arranger": Object {
                    "attending": Array [
                      "1",
                      "2",
                    ],
                    "email": "tjalle@yahoo.com",
                    "firstName": "Tjalle",
                    "id": "1",
                    "lastName": "Bergkvist",
                    "password": "grillkorv",
                    "reviewed": Array [
                      "1",
                    ],
                  },
                  "attending": 54,
                  "capacity": null,
                  "comments": Array [
                    Object {
                      "id": "1",
                      "submittedAt": 2021-12-16T14:58:33.837Z,
                      "text": "Great day!",
                      "userId": "2",
                    },
                    Object {
                      "id": "2",
                      "submittedAt": 2021-12-16T14:59:22.526Z,
                      "text": "Yes, it was, thanks!",
                      "userId": "1",
                    },
                    Object {
                      "id": "3",
                      "submittedAt": 2021-12-17T12:34:42.949Z,
                      "text": "Damn, I missed this",
                      "userId": null,
                    },
                  ],
                  "date": 2021-08-28T17:00:00.000Z,
                  "description": "We meet in the park with pizzas and hang out all night",
                  "id": "1",
                  "location": Object {
                    "city": "Stockholm",
                    "street": "Vitabergsparken",
                  },
                  "online": false,
                  "reviews": Array [
                    Object {
                      "id": "1",
                      "meetupId": "1",
                      "rating": 4,
                    },
                    Object {
                      "id": "2",
                      "meetupId": "1",
                      "rating": 5,
                    },
                  ],
                  "title": "Pizza Picnic",
                },
                Object {
                  "arranger": Object {
                    "attending": Array [],
                    "email": "babsan@yahoo.com",
                    "firstName": "Barbro",
                    "id": "3",
                    "lastName": "Lundborg",
                    "password": "password123",
                    "reviewed": Array [],
                  },
                  "attending": 356,
                  "capacity": 500,
                  "comments": Array [],
                  "date": 2021-10-31T18:00:00.000Z,
                  "description": "Halloween party in the street, all night long",
                  "id": "5",
                  "location": Object {
                    "city": "Stockholm",
                    "street": "Strandvägen",
                  },
                  "online": false,
                  "reviews": Array [],
                  "title": "Halloween Street Party",
                },
              ]
            }
          >
            <ul
              className="meetupList"
              data-testid="past-events"
            >
              <MeetupCard
                key="1"
                meetup={
                  Object {
                    "arranger": Object {
                      "attending": Array [
                        "1",
                        "2",
                      ],
                      "email": "tjalle@yahoo.com",
                      "firstName": "Tjalle",
                      "id": "1",
                      "lastName": "Bergkvist",
                      "password": "grillkorv",
                      "reviewed": Array [
                        "1",
                      ],
                    },
                    "attending": 54,
                    "capacity": null,
                    "comments": Array [
                      Object {
                        "id": "1",
                        "submittedAt": 2021-12-16T14:58:33.837Z,
                        "text": "Great day!",
                        "userId": "2",
                      },
                      Object {
                        "id": "2",
                        "submittedAt": 2021-12-16T14:59:22.526Z,
                        "text": "Yes, it was, thanks!",
                        "userId": "1",
                      },
                      Object {
                        "id": "3",
                        "submittedAt": 2021-12-17T12:34:42.949Z,
                        "text": "Damn, I missed this",
                        "userId": null,
                      },
                    ],
                    "date": 2021-08-28T17:00:00.000Z,
                    "description": "We meet in the park with pizzas and hang out all night",
                    "id": "1",
                    "location": Object {
                      "city": "Stockholm",
                      "street": "Vitabergsparken",
                    },
                    "online": false,
                    "reviews": Array [
                      Object {
                        "id": "1",
                        "meetupId": "1",
                        "rating": 4,
                      },
                      Object {
                        "id": "2",
                        "meetupId": "1",
                        "rating": 5,
                      },
                    ],
                    "title": "Pizza Picnic",
                  }
                }
              >
                <li
                  aria-label="meetup-card"
                  className="cardWrapper"
                >
                  <div
                    className="cardHeader"
                    style={
                      Object {
                        "backgroundColor": "#DCDDDE",
                      }
                    }
                    title="past-event-header"
                  >
                    <h2>
                      Pizza Picnic
                    </h2>
                  </div>
                  <div
                    className="cardBody"
                  >
                    <div
                      className="leftColumn"
                    >
                      <div
                        className="dateRow"
                      >
                        <img
                          alt="Time icon"
                          className="icon"
                          src="time.png"
                        />
                        <h3
                          title="date"
                        >
                          28 August 2021, 19:00
                           
                          CET
                        </h3>
                      </div>
                      <div
                        className="locationRow"
                      >
                        <img
                          alt="Map pin icon"
                          className="icon"
                          src="pin.png"
                        />
                        <h3
                          title="location"
                        >
                          Vitabergsparken, Stockholm
                        </h3>
                      </div>
                      <p
                        className="capacityRow"
                        title="capacity"
                      >
                        <strong>
                          Capacity: 
                        </strong>
                        <span
                          className="available"
                        >
                          Unlimited
                        </span>
                      </p>
                    </div>
                    <div
                      className="rightColumn"
                    >
                      <Link
                        to="/meetup/1"
                      >
                        <a
                          href="/meetup/1"
                          onClick={[Function]}
                        >
                          <img
                            alt="Arrow icon"
                            className="arrowIcon"
                            src="arrow.png"
                            title="See meetup details"
                          />
                        </a>
                      </Link>
                    </div>
                  </div>
                </li>
              </MeetupCard>
              <MeetupCard
                key="5"
                meetup={
                  Object {
                    "arranger": Object {
                      "attending": Array [],
                      "email": "babsan@yahoo.com",
                      "firstName": "Barbro",
                      "id": "3",
                      "lastName": "Lundborg",
                      "password": "password123",
                      "reviewed": Array [],
                    },
                    "attending": 356,
                    "capacity": 500,
                    "comments": Array [],
                    "date": 2021-10-31T18:00:00.000Z,
                    "description": "Halloween party in the street, all night long",
                    "id": "5",
                    "location": Object {
                      "city": "Stockholm",
                      "street": "Strandvägen",
                    },
                    "online": false,
                    "reviews": Array [],
                    "title": "Halloween Street Party",
                  }
                }
              >
                <li
                  aria-label="meetup-card"
                  className="cardWrapper"
                >
                  <div
                    className="cardHeader"
                    style={
                      Object {
                        "backgroundColor": "#DCDDDE",
                      }
                    }
                    title="past-event-header"
                  >
                    <h2>
                      Halloween Street Party
                    </h2>
                  </div>
                  <div
                    className="cardBody"
                  >
                    <div
                      className="leftColumn"
                    >
                      <div
                        className="dateRow"
                      >
                        <img
                          alt="Time icon"
                          className="icon"
                          src="time.png"
                        />
                        <h3
                          title="date"
                        >
                          31 October 2021, 19:00
                           
                          CET
                        </h3>
                      </div>
                      <div
                        className="locationRow"
                      >
                        <img
                          alt="Map pin icon"
                          className="icon"
                          src="pin.png"
                        />
                        <h3
                          title="location"
                        >
                          Strandvägen, Stockholm
                        </h3>
                      </div>
                      <p
                        className="capacityRow"
                        title="capacity"
                      >
                        <strong>
                          Capacity: 
                        </strong>
                        <span>
                          500, 
                        </span>
                        <span
                          className="available"
                        >
                          144 available
                        </span>
                      </p>
                    </div>
                    <div
                      className="rightColumn"
                    >
                      <Link
                        to="/meetup/5"
                      >
                        <a
                          href="/meetup/5"
                          onClick={[Function]}
                        >
                          <img
                            alt="Arrow icon"
                            className="arrowIcon"
                            src="arrow.png"
                            title="See meetup details"
                          />
                        </a>
                      </Link>
                    </div>
                  </div>
                </li>
              </MeetupCard>
            </ul>
          </MeetupList>
        </section>
      </Home>
    </Routes>
  </div>
</App>
`;
